src/EditorFeatures/CSharpTest/CodeActions/Preview/PreviewTests.cs

```diff
@@ -102,7 +103,7 @@ public void TestPickTheRightPreview_NoPreference()
                 GetMainDocumentAndPreviews(workspace, out document, out previews);
 
                 // The changed document comes first.
-                var preview = previews.TakeNextPreview();
+                var preview = previews.TakeNextPreviewAsync().PumpingWaitResult();
```

Comment:
You mean tests [like this](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/54db2faa6b3493da0830511ab3153106cde1c36b/StyleCop.Analyzers/StyleCop.Analyzers.Test/LayoutRules/SA1516UnitTests.cs#L373-L374)?

It does support `async` tests, but unfortunately does not pass in a `CancellationToken` yet (xunit/xunit#345).